Dockerize korotagger and automate DB Initialization

JakeFrosty opened the pull request.

JakeFrosty committed:
Update .gitmodules

JakeFrosty committed:
Dockerize app and make DB Init automatic

Yarn commented on .gitmodules:
```diff
@@ -1,3 +1,3 @@
 [submodule "discord_lib"]
 	path = discord_lib
-	url = git@github.com:Yarn/discord-lib.git
+	url = https://github.com/Yarn/discord-lib.git
```
remove this change, using an http url would prevent me from being able to push changes

Yarn commented on entrypoint.sh:
```diff
@@ -0,0 +1,62 @@
+    for sql_file in /app/migrations/*.sql; do
+        if [ -f "$sql_file" ]; then
+            echo "Running SQL script $sql_file..."
+            PGPASSWORD="$POSTGRES_PASSWORD" psql -h "$POSTGRES_HOST" -p "$POSTGRES_PORT" -U "$POSTGRES_USER" -d "$POSTGRES_DB" -f "$sql_file"
```
`/app/korotagger migrate` should be run before each launch instead of using psql. the migrations are included in the binary so copying the sql files into tho container isn't necessary. it creates a table to track which migrations are applied so the check_first_time_run function is also not needed.

Yarn commented on docker-compose.yaml:
```diff
@@ -0,0 +1,39 @@
+services:
+  kt-app:
+     image: jakefrosty/korotagger:1.3
```
could you add the docker command to generate the image and start docker compose to the readme

JakeFrosty replied:
Will do! I'll make a follow up commit to the README.md file

Yarn commented on docker-compose.yaml:
```diff
@@ -0,0 +1,39 @@
+     mem_limit: 100m
+     container_name: korotagger-app
+     environment:
+       yt_api_key: your-yt-api-key
```
could you list the environment variable a user would be expected to set in the readme, this would be useful for people running outside docker as well

JakeFrosty replied:
I will make a new make an updated README.md sure!
I'll include syntax for directly running and via docker aswell

Yarn commented on docker-compose.yaml:
```diff
@@ -0,0 +1,39 @@
+     environment:
+       yt_api_key: your-yt-api-key
+       holodex_api_key: your-holodex-api-key
+       POSTGRES_HOST: kt-db
```
is there a reason to split this out into multiple environment variables over providing an example for how to format a postgres url?

there doesn't seem to be any handling for special characters or such so this would presumably have the same pitfalls, just less transparently

JakeFrosty replied:
The community docker-compose files I'm used to have always used split database credentials and I didn't wanna make it different

JakeFrosty committed:
Update .gitmodules

Revert https to git

JakeFrosty committed:
Update with build directive instead of image

replace image with build directive, remove unecessary cpu and memory limits and leave it up to user to add them back

JakeFrosty committed:
Update entrypoint.sh

remove check for my own inserted "config.first_time_run", instead use "member.required_roles" which is the last created table on the third SQL script at migrations

JakeFrosty committed:
Update docker-compose.yaml

comment out image directive, encourage building instead for the latest builds from repo directly.

add double quotes on example postgres password to encourage properly quoting password

Yarn commented on entrypoint.sh:
```diff
@@ -0,0 +1,50 @@
+wait_for_postgres
+
+# Check if it is the first time run
+if check_first_time_run; then
```
should remove check_first_time_run. korotagger migrate will only apply new migrations already and we probably want new migrations applied after updates

Yarn commented:
sorry for the slow response, i think there's just the one thing left to fix.

it probably makes sense to leave the 100MB memory limit for korotagger. korotagger usually only uses around 12MB at most.
